app/controllers/landing_page_controller.rb

```diff
@@ -183,12 +191,8 @@ def parse_int(int_str_or_nil)
     nil
   end
 
-  def community_id(request)
-    request.env[:current_marketplace]&.id
-  end
-
-  def community_default_locale(request)
-    request.env[:current_marketplace]&.default_locale
+  def current_community(request)
+    @current_community ||= request.env[:current_marketplace]
```

Comment:
In any case, it would be a good idea to merge these two in first and then continue with the extracting as a separate PR so we can ensure we have something ready to go.